In [1]:
from neural_network.binary_nn import NN
from datasets.sipm_dataset import SiPMDataset
from genetic_algorithm.algorithm import GeneticAlgorithm

In [2]:
# GA parameters
genome_length = 128  # smaller genome for quick test
mutation_prob = 0.05
pop_size = 10       # small population for demo
ngen = 5            # few generations

nn = NN((genome_length, 64, 128, 2), input=SiPMDataset(n_samples=genome_length))

ga = GeneticAlgorithm(
    nn=nn, 
    genome_length=genome_length,
    mutation_prob=mutation_prob,
    pop_size=pop_size,
    ngen=ngen,
    elite_size=2
)

TypeError: GeneticAlgorithm.__init__() got an unexpected keyword argument 'genome_length'

In [ ]:
pop, log, hof = ga.run()

Create init population...
Start evolution...


ValueError: cannot reshape array of size 128 into shape (64,128,2)

funktioniert fuer alle layer aus letzten -> 2 (ValueError: operands could not be broadcast together with shapes (2,) (64,128) )

In [ ]:
print("Best individual:", hof[0])
print("Best fitness:", ga.nn.fitness(hof[0]))
print("GA run complete.\n")
# Example run for the HCL4BNN package
print(f"log: \n{log}")
print(f"pop: \n{pop}")

# Testing

In [ ]:
nn = NN((2, 8, 2), input=SiPMDataset())
x, y = SiPMDataset().load_data()
print(f"Inputs (X): \n{x}\n\n\nLabels (y): \n{y}\n\n\n")